doc/mellanox/sfp-error-status.md

```diff
@@ -0,0 +1,195 @@
+# Fetch SFP error status through CLI #
+
+## Table of Content
+
+### Revision
+
+### Scope
+
+This is the high level design for the feature "fetch SFP error status through CLI".
+
+### Definitions/Abbreviations
+
+This section covers the abbreviation if any, used in this high-level design document and its definitions.
+xcvrd: Transceiver daemon
+CLI: Command line interface
+
+### Overview
+
+The purpose of this section is to give an overview of high-level design document and its architecture implementation in SONiC.
+
+### Requirements
+
+The requirement:
+
+- To expose the SFP error status to CLI
+- One shot command for debugging, which means the error status should be directly polled from low lever, eg. SDK, instead of being cached in the database.
+- Only supported on Mellanox system. The command will be under `show platform mlnx`
+
+### Architecture Design
+
+CLI to call platform API directly and platform API to fetch error status by calling SDK API.
+
+- A new platform API is introduced to fetch the error code of the SPF module via using the SDK API.
+- A CLI command is introduced to display the error status to the user. Since the platform API is available in pmon docker only, the CLI has to call platform API via `docker exec` and then parse the output of the platform API.
+- No database change is required.
+
+### High-Level Design
+
+#### Platform API
+
+A new platform API in class SFP is required for polling the error status of the SFP module.
+
+```
+def get_error_status(self)
+    """
+    Get error status of the SFP module
+    Returns:
+        string: represent the error code
+    """
+```
+
+It calls the SDK API and translates the return code to a human-readable string. In case the SFP module isn't plugged-in, it will return "no error".
+
+The possible errors are listed below:
+
+- Power budget exceeded
+- Long range
+- Bus stuck
+- Bad unsupported eeprom
+- Enforce part number list
+- Unsupported cable
+- High temperature
+- Bad cable
+- PMD type not enabled
+- PCIE system power slot exceeded
+
+#### CLI
+
+##### CLI syntax
+
+The command `show platform mlnx transceiver error-status` is designed to fetch and display the error status of the SFP modules. It is implemented by calling the platform API.
+
+```
+sonic#show platform mlnx transceiver error-status --port <port_name>
+```
+
+If the parameter `--port` is provided, the command will display the error status of the port. Otherwise, it will display the error status of all the ports.
+
+```
+admin@sonic:~$ show platform mlnx transceiver error-status --port Ethernet8
+Port       Error Status
+---------  ------------------------------------
+Ethernet8  No error
+
+admin@sonic:~$ show platform mlnx transceiver error-status 
+Port         Error Status
+-----------  ----------------------------------------------
+Ethernet0    No error
+Ethernet4    No error
+Ethernet8    No error
+Ethernet12   No error
```

Comment:
is it possible to have multi errors? can we give an example if yes?  can we give an example if SFP is not present?

Reply:
Only one error is supported for each module.
If SFP is not present, the error code should be "OK" as well.